lib/module.js

```diff
@@ -219,17 +219,25 @@ if (process.platform === 'win32') {
     // note: this approach *only* works when the path is guaranteed
     // to be absolute.  Doing a fully-edge-case-correct path.split
     // that works on both Windows and Posix is non-trivial.
+
+    // return root node_modules when path is 'D:\\'.
+    // path.resolve will make sure from.length >=3 in Windows.
+    if (from[from.length - 1] === '\\' && from[from.length - 2] === ':')
```

Comment:
What about prepending a `from.length >= 2` check here and then changing the string equality checks to `.charCodeAt()`-based checks? The length check will help prevent a deopt when accessing a non-existent index and the character comparison changes should improve performance a little bit.